Update setup for clang

CasualPokePlayer opened the pull request:
Cherrypicks in http://git.musl-libc.org/cgit/musl/commit/?id=c40157d87ef0b585a90ff5230fdf69cddc824d7f

See https://github.com/TASEmulators/BizHawk/pull/3529

CasualPokePlayer committed:
changes for building waterbox with clang / llvm tools instead of gcc and co

CasualPokePlayer committed:
go back to full -nostdinc

CasualPokePlayer committed:
compat with both gcc and clang

Morilli commented on wbox_configure.sh:
```diff
@@ -1,9 +1,9 @@
-export LDFLAGS=""
-export CFLAGS="-Werror -fvisibility=hidden -mcmodel=large -mstack-protector-guard=global -no-pie -fno-pic -fno-pie" # TODO: Add -flto
-export AR="gcc-ar"
-export RANLIB="gcc-ranlib"
+export CFLAGS="-Werror -fvisibility=hidden -mcmodel=large -mstack-protector-guard=global -fno-pic -fno-pie -fcf-protection=none" # TODO: Add -flto
+export AR="llvm-ar"
+export RANLIB="llvm-ranlib"
```
Could this be switched by user `CC` choice as well? Aka using gcc-ar for gcc and llvm-ar for clang? Or is there a reason linking is now always done with llvm?

CasualPokePlayer replied:
-ar and -ranlib really don't matter in any case, they just archive the objects into .a files. llvm-ar and llvm-ranlib would be more "guaranteed" to exist given the waterbox setup requires llvm package to be installed. Although I suppose at that point just let autotools figure out what ar/ranlib is available and don't try to explicitly set them.

Morilli replied:
Fwiw on my ubuntu install, just installing `clang` installs `llvm-15` as a dependency, however that only provides `llvm-ar-15` (etc.). I did not have the `llvm` package explicitly installed, which would have provided those symlinks (which is arguably my fault because the readme does list `llvm` as a requirement).

CasualPokePlayer committed:
Remove AR/RANLIB specifications

These don't really matter in any case (they just archive .o fies, how can you screw that up?) configure will figure out whatever ar/ranlib is available and use that

CasualPokePlayer committed:
fixup musl-clang script, again
Wunused-command-line-argument's output makes sense here kind of but honestly a very annoying default

CasualPokePlayer committed:
revert ab9563afdde1d7acb2200b3bcbb7d206458e9485

I don't like this, but it seems to be required sadly...
